src/System/Taffybar/Widget/Generic/Icon.hs

```diff
@@ -49,6 +63,37 @@ pollingIconImageWidgetNew path interval cmd = liftIO $ do
     return ()
   putInBox icon
 
+-- | Create a new widget that updates itself at regular intervals.  The
+-- function
+--
+-- > pollingIconImageWidgetNewFromName name interval cmd
+--
+-- returns a widget with initial icon whose name is @name@.  The widget
+-- forks a thread to update its contents every @interval@ seconds.
+-- The command should return the name of a valid icon.
+--
+-- If the IO action throws an exception, it will be swallowed and the
+-- label will not update until the update interval expires.
+pollingIconImageWidgetNewFromName
+  :: MonadIO m
+  => T.Text    -- ^ Icon Name
+  -> Double    -- ^ Update interval (in seconds)
+  -> IO T.Text -- ^ Command to run update the icon name
+  -> m Widget
+pollingIconImageWidgetNewFromName name interval cmd = liftIO $ do
+  icon <- imageNewFromIconName (Just name) (fromIntegral $ fromEnum IconSizeMenu)
+  _ <- onWidgetRealize icon $ do
+    _ <- forkIO $ forever $ do
+      let tryUpdate = do
+            name' <- cmd
+            postGUIASync $
+              imageSetFromIconName icon (Just name') $
+                fromIntegral $ fromEnum IconSizeMenu
+      E.catch tryUpdate ignoreIOException
+      threadDelay $ floor (interval * 1000000)
+    return ()
+  putInBox icon
```

Comment:
This repeats too much code from the above function. The only lines that are different here are 90 and 84. It should be pretty easy to factor those part out so they are passed in as a parameter to a more general function that can handle both cases. You can keep the new functions names as simple aliases that just call the more general function appropriately.

Reply:
Will do!

Reply:
Resolved via ec2f227